php/commands/src/CLI_Command.php

```diff
@@ -103,6 +103,10 @@ public function info( $_, $assoc_args ) {
 
 		$runner = WP_CLI::get_runner();
 
+		$system_os           = php_uname( 's' );
+		$system_architecture = php_uname( 'm' );
+		$current_shell       = ( isset( $_SERVER['SHELL'] ) ) ? $_SERVER['SHELL'] : '';
```

Comment:
I think just `$shell` for the variable - "current" doesn't add anything. Also could you use `getenv()` instead and for Windows use `ComSpec` if `SHELL` not available:
```
$shell = getenv( 'SHELL' );
if ( ! $shell && Utils\is_windows() ) {
	$shell = getenv( 'ComSpec' );
}
```

Reply:
thanks @gitlost for this
```
if ( ! $shell && Utils\is_windows() ) {
	$shell = getenv( 'ComSpec' );
}
```
i didn't know about this.

Reply:
Yeah, I've become a bit of a Windows guru over the last few weeks (don't use it locally or for servers)! `ComSpec` always returns path to `cmd.exe`, even on PowerShell, so isn't that great but looks a bit better anyway...

Reply:
I tested the code on Windows 7 ( installed VirtualBox just for that ) works perfectly:100:

Reply:
Cool, thanks @thrijith !